src/ensemble_scheduler/ensemble_scheduler.cc

```diff
@@ -1033,6 +1040,50 @@ EnsembleContext::ReshapeTensorDims(
   return res;
 }
 
+// Caching function
+void
+EnsembleContext::CacheEnsembleTopLevelRequest(
+    std::unique_ptr<InferenceResponse>& response)
```

Comment:
If the model is a non-decoupled model that sends the final flag and response separately is it correctly handled here?

Reply:
I'm gonna file a ticket for separate investigation on this flow for both existing caching (dynamic_batch_scheduler) and this new ensemble caching if that's good with you @Tabrizian.

Reply:
Sounds good!